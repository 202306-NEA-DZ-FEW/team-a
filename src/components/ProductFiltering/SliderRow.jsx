import { MdChevronLeft, MdChevronRight } from "react-icons/md";

import CategoryCard from "./CategoryCard";

function SliderRow({ title, categories, rowID, queryParams }) {
  const slideLeft = () => {
    const slider = document.getElementById("slider" + rowID);
    slider.scrollLeft = slider.scrollLeft - 600;
  };
  const slideRight = () => {
    const slider = document.getElementById("slider" + rowID);
    slider.scrollLeft = slider.scrollLeft + 600;
  };

  return (
    <section className='flex flex-col pt-8 gap-2 bg-black'>
      {title && (
        <h2 className='font-bold text-white text-2xl md:text-3xl text-center'>
          {title}
        </h2>
      )}
      <div className='relative flex items-center group'>
        <MdChevronLeft
          onClick={slideLeft}
          className='bg-white left-0 rounded-full absolute opacity-50 hover:opacity-100 cursor-pointer z-10 hidden group-hover:block'
          size={40}
        />
        <div
          id={"slider" + rowID}
          className='w-full py-10 flex gap-8 px-8 overflow-x-scroll scroll-smooth no-scrollbar'
        >
          {categories.map((category) => (
            <CategoryCard
              key={category.dataKey}
              queryParams={queryParams}
              link={category.dataKey}
              title={category.name}
              imageUrl={category.imageURL}
            />
          ))}
        </div>
        <MdChevronRight
          onClick={slideRight}
          className='bg-white right-0 rounded-full absolute opacity-50 hover:opacity-100 cursor-pointer z-20 hidden group-hover:block'
          size={40}
        />
      </div>
    </section>
  );
}

export default SliderRow;
