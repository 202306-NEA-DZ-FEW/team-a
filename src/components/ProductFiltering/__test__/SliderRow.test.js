import image from "public/images/mepi1.png";
import renderer from "react-test-renderer";

import SliderRow from "../SliderRow";

it("renders correctly", () => {
  const mockItems = [
    {
      id: 0,
      name: "name",
      description: "description",
      dataKey: "dataKey",
      imageURL: image,
    },
  ];
  const queryParams = {};
  const title = "title";
  const tree = renderer
    .create(
      <SliderRow
        title={title}
        categories={mockItems}
        rowID={1}
        queryParams={queryParams}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
