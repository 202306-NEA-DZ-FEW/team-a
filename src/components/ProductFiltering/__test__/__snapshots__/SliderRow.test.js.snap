// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="flex flex-col pt-8 gap-2 bg-black"
>
  <h2
    className="font-bold text-white text-2xl md:text-3xl text-center"
  >
    title
  </h2>
  <div
    className="relative flex items-center group"
  >
    <svg
      className="bg-white left-0 rounded-full absolute opacity-50 hover:opacity-100 cursor-pointer z-10 hidden group-hover:block"
      fill="currentColor"
      height={40}
      onClick={[Function]}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width={40}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
      />
    </svg>
    <div
      className="w-full py-10 flex gap-8 px-8 overflow-x-scroll scroll-smooth no-scrollbar"
      id="slider1"
    >
      <div
        aria-label="name"
        className="card w-60 h-40 card-compact border-none relative rounded-3x"
      >
        <a
          href="/products?category=dataKey"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          <figure
            className="w-56 h-40 relative rounded-3xl shadow-xl"
          >
            <img
              alt="name"
              className="object-cover hover:scale-125 transition-all duration-700"
              data-nimg="fill"
              decoding="async"
              fetchpriority="high"
              onError={[Function]}
              onLoad={[Function]}
              sizes="(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw"
              src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
              srcSet="/_next/image?url=%2Fimg.jpg&w=256&q=75 256w, /_next/image?url=%2Fimg.jpg&w=384&q=75 384w, /_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
              style={
                Object {
                  "bottom": 0,
                  "color": "transparent",
                  "height": "100%",
                  "left": 0,
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": "100%",
                }
              }
            />
            <h3
              className="text-white text-center absolute bottom-[10%] px-2 py-1 w-[90%] text-md card-bordered rounded-xl bg-black bg-opacity-30"
            >
              name
            </h3>
          </figure>
        </a>
      </div>
    </div>
    <svg
      className="bg-white right-0 rounded-full absolute opacity-50 hover:opacity-100 cursor-pointer z-20 hidden group-hover:block"
      fill="currentColor"
      height={40}
      onClick={[Function]}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width={40}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
      />
    </svg>
  </div>
</section>
`;
